poutyne/framework/model.py

```diff
@@ -763,10 +787,26 @@ def _compute_loss_and_metrics(self, x, y, return_loss_tensor=False, return_pred=
         return loss, metrics, pred_y
 
     def _compute_metrics(self, pred_y, y):
-        return np.array([float(metric(pred_y, y)) for metric in self.batch_metrics])
+        metrics = [metric(pred_y, y) for metric in self.batch_metrics]
+        return self._compute_metric_array(metrics, self.unflatten_batch_metrics_names)
 
     def _get_epoch_metrics(self):
-        return np.array([float(epoch_metric.get_metric()) for epoch_metric in self.epoch_metrics])
+        metrics = [epoch_metric.get_metric() for epoch_metric in self.epoch_metrics]
+        return self._compute_metric_array(metrics, self.unflatten_epoch_metrics_names)
+
+    def _compute_metric_array(self, metrics_list, names_list):
+        def _get_metric(names, metrics):
+            names = [names] if isinstance(names, str) else names
+            if (torch.is_tensor(metrics) or isinstance(metrics, np.ndarray)) and len(metrics.shape) == 0:
+                return [float(metrics)]
+            if isinstance(metrics, Mapping):
```

Comment:
```suggestion
            elif isinstance(metrics, Mapping):
```

Reply:
Pylint gives a warning in that case because there is a return statement inside the if.

Reply:
Ah true. I'm not a fanatic of multiples returns also. I think it's more clear to use the same variable name for all the case in a if-else and return that variable.

Reply:
Alright, I pushed something.